src/main/java/com/google/sps/servlets/BusLocations.java

```diff
@@ -0,0 +1,98 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.sps.data.CoordinateCalculator;
+import com.google.sps.data.EntityType;
+import com.google.sps.data.Office;
+import com.google.sps.data.OfficeManager;
+import com.google.sps.data.QueryHelper;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles adding and retreiving listings & locations */
+@WebServlet("/busLocations")
+public class BusLocations extends HttpServlet {
+
+  private final int numStops = 10;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String office = request.getParameter("office");
+    Office selectedOffice = OfficeManager.offices.get(office);
+    double distanceInKilometers = CoordinateCalculator.milesToKilometers(2.3);
+
+    List<Entity> entityList =
+        QueryHelper.getDistanceBasedEntities(
+            EntityType.BUS_STOP, numStops, selectedOffice, distanceInKilometers);
+
+    List<Entity> filteredEntityList =
+        CoordinateCalculator.filterOutOfRangeLatitudeEntities(
+            distanceInKilometers,
+            selectedOffice.getLatitude(),
+            selectedOffice.getLongitude(),
+            entityList);
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(filteredEntityList));
+  }
+
+  @Override // Creates a new BusStop
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    createListing(request);
+  }
+
+  private void createListing(HttpServletRequest request) throws IOException {
+
+    // To be used for timestamp
+    long timestamp = System.currentTimeMillis();
+
+    // Create JSON object with GSON
+    String requestData = request.getReader().lines().collect(Collectors.joining());
+    JsonObject stopJson = new Gson().fromJson(requestData, JsonObject.class);
+
+    // Creating DataStore Entity
+    Entity taskEntity = new Entity(EntityType.BUS_STOP.getValue());
+    taskEntity.setProperty("timestamp", timestamp);
```

Comment:
This probably reflect my ignorance about data store 😂 : is the timestamp used to create index?

Reply:
It's a good question actually! I kept the timestamp because I figure keeping it wouldn't really affect things in terms of performance,  and I figured it MIGHT be useful at some point for something in the future... though at the moment it serves no purpose. Perhaps I should just get rid of it. 

As far indexing goes, I think DataStore automatically creates a key for every Entity. Though while I was reading a bit about DataStore, I think I read almost every attribute of an Entity is indexed (excluding some particular values), unless specified otherwise. 

https://cloud.google.com/datastore/docs/concepts/indexes